client/components/fixed-navigation-header/index.tsx

```diff
@@ -55,16 +56,50 @@ interface Props {
 	className?: string;
 	children?: ReactNode;
 	navigationItems: { label: string; href?: string }[];
+	contentRef?: React.RefObject< HTMLElement >;
 }
 
 const FixedNavigationHeader: React.FunctionComponent< Props > = ( props ) => {
-	const { id, className, children, navigationItems } = props;
+	const { id, className, children, navigationItems, contentRef } = props;
+	const actionsRef = useRef< HTMLDivElement >( null );
+	const headerRef = useRef< HTMLElement >( null );
+
+	useEffect( () => {
+		if ( ! contentRef ) {
+			return;
+		}
+
+		const handleScroll = () => {
+			const headerHeight = headerRef?.current?.getBoundingClientRect().height;
+			const offset =
+				contentRef.current && headerHeight ? contentRef.current.offsetTop - headerHeight : 0;
+			const scrollPosition = window.scrollY;
+			const actionElement = actionsRef?.current;
+
+			if ( ! actionElement ) {
+				return;
+			}
+
+			if ( offset > 0 && scrollPosition < offset ) {
+				actionElement.style.visibility = 'hidden';
+			} else {
+				actionElement.style.visibility = 'visible';
+			}
+		};
+
+		handleScroll();
+
+		window.addEventListener( 'scroll', handleScroll );
+		return () => {
+			window.removeEventListener( 'scroll', handleScroll );
+		};
+	}, [ contentRef ] );
```

Comment:
After looking at https://github.com/Automattic/wp-calypso/pull/62080/files, I found at about this code. Can we add some comments about what this `useEffect` does for posterity? Also, I noticed a new prop which should be added in the `readme` https://github.com/Automattic/wp-calypso/blob/831b4c593e979f6b214724ff4693c2aebbc9fa26/client/components/fixed-navigation-header/README.md#L25-L29 

Moreover, it seems to me that it could have been the parent component responsibility to render or not the `actionElement` (perhaps by using a similar `useEffect` rather than having to pass the `contentRef`. Is there a reason for choosing to place this functionality here?

Reply:
@cpapazoglou Would you mind opening issues for these so we can triage them, rather than commenting on a merged PR?

Reply:
Sure, I just thought we can discuss it first. Added here https://github.com/Automattic/wp-calypso/issues/62166